repos/spack_repo/builtin/packages/ddc/package.py

```diff
@@ -0,0 +1,147 @@
+from spack_repo.builtin.build_systems.rocm import ROCmPackage
+
+from spack.package import *
+
+
+class Ddc(CMakePackage):
+    """DDC is the discrete domain computation library."""
+
+    homepage = "https://github.com/CExA-project/ddc"
+    git = "https://github.com/CExA-project/ddc.git"
+    url = "https://github.com/CExA-project/ddc/archive/refs/tags/v0.4.2.tar.gz"
+
+    maintainers("tpadioleau")
+
+    license("MIT", checked_by="tpadioleau")
+
+    version("develop", branch="main", no_cache=True)
+    version("0.8.0", sha256="6c6d28f1d406e1417021f88d748829cae0afce2cb3714cf82fd3f4cd3b7b91b4")
+    version("0.7.0", sha256="128dd93d0021da35dcd62db7eabab3136c826a924dbe90368361d347e6bd3111")
+
+    variant("fft", default=True, description="Build DDC kernels for FFT")
+    variant("pdi", default=True, description="Build DDC PDI wrapper")
+    variant("splines", default=True, description="Build DDC kernels for splines")
+    variant("deprecated_code", default=True, description="Build deprecated code")
+    variant(
+        "double_precision",
+        default=True,
+        description="Use double precision floating point numbers instead of single precision",
+    )
+    variant("tests", default=False, description="Build the tests")
+
+    depends_on("cxx", type="build")
+
+    depends_on("cmake@3.22:3", type="build")
+    depends_on("kokkos@4.4.1:4")
+
+    with when("+fft"):
+        for variant, backend in [
+            ("~openmp", "host_backend=fftw-serial"),
+            ("+openmp", "host_backend=fftw-openmp"),
+            ("+cuda", "device_backend=cufft"),
+            ("+rocm", "device_backend=hipfft"),
+            ("+sycl", "device_backend=onemkl"),
+        ]:
+            depends_on(f"kokkos-fft@0.3.0 {backend}", when=f"^kokkos {variant}")
+
+    with when("+splines"):
+        depends_on("ginkgo@1.8:1")
+        depends_on("kokkos-kernels@4.5.1:4")
+        depends_on("blas")
+        depends_on("lapack")
+
+        for arch in CudaPackage.cuda_arch_values:
+            with when(f"^kokkos +cuda cuda_arch={arch}"):
+                requires(f"^ginkgo +cuda cuda_arch={arch}")
+                requires(f"^kokkos-kernels +cuda cuda_arch={arch}")
+
+        for target in ROCmPackage.amdgpu_targets:
+            requires(
+                f"^ginkgo +rocm amdgpu_target={target}",
+                when=f"^kokkos +rocm amdgpu_target={target}",
+            )
+
+        requires("^ginkgo +sycl", when="^kokkos +sycl")
+        requires("^ginkgo +openmp", when="^kokkos +openmp")
+
+        # We cannot build shared libraries when relocatable device code is used
+        for backend in ["cuda", "rocm", "sycl"]:
+            requires("^kokkos-kernels ~shared", when=f"^kokkos +{backend}")
+
+    depends_on("pdi@1.6:1", when="+pdi")
+
+    with when("+tests"):
+        depends_on("googletest@1.14:1 +gmock")
+        depends_on("pdiplugin-user-code +tests", type=("build", "test"), when="+pdi")
```

Comment:
```suggestion
        depends_on("pdiplugin-user-code@1.6:1", type=("build", "test"), when="+pdi")
```

I don't remember why we were relying on `+tests` ?

Reply:
That's a mistake on my side, it's not needed actually